src/Components/Server/src/Circuits/PostConfigureCircuitOptionsCircuitIdDataprotector.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Components.Server.Circuits
+{
+    internal class PostConfigureCircuitOptionsCircuitIdDataprotector : IPostConfigureOptions<CircuitOptions>
+    {
+        private const string CircuitIdProtectorPurpose = "Microsoft.AspNetCore.Components.Server";
+        private readonly IDataProtectionProvider _dataProtectionProvider;
+
+        public PostConfigureCircuitOptionsCircuitIdDataprotector(IDataProtectionProvider dataProtectionProvider)
+        {
+            _dataProtectionProvider = dataProtectionProvider;
+        }
+        public void PostConfigure(string name, CircuitOptions options)
+        {
+            options.CircuitIdProtector ??= _dataProtectionProvider.CreateProtector(CircuitIdProtectorPurpose);
```

Comment:
This is how other parts of the stack initialize the protector

Reply:
Yup, you're just copying session and that's the right way to go about it.